javascript/app-discover/src/components/MenuBar.hooks.tsx

```diff
@@ -207,26 +176,99 @@ export function useFixedInterventionRangesToggleMenuItem() {
 	)
 }
 
-export function useTogglePauseButtonMenuItem() {
-	const [pauseAutoRun, setPauseAutoRun] = useRecoilState(PauseAutoRunState)
-	const paused = pauseAutoRun === CausalDiscoveryAlgorithm.None
-	const togglePaused = useCallback(() => {
-		setPauseAutoRun(paused ? undefined : CausalDiscoveryAlgorithm.None)
-	}, [setPauseAutoRun, paused])
+export function useAutoLayoutToggleMenuItem() {
+	const [autoLayoutEnabled, setAutoLayoutEnabled] = useRecoilState(
+		AutoLayoutEnabledState,
+	)
+	const handleClick = useRecoilCallback(
+		({ snapshot, set }) =>
+			async () => {
+				set(AutoLayoutEnabledState, true)
+				const autoLayoutSnapshot = await snapshot.getPromise(
+					AutoLayoutEnabledState,
+				)
+				if (autoLayoutSnapshot) {
+					setTimeout(() => {
+						set(AutoLayoutEnabledState, false)
+					}, 10)
```

Comment:
why 10ms? Would zero work? Do we just want to process this in the next tick?

Reply:
This was refactored and the setTimeout was removed